### PR TITLE
feat(uat): add T20 scenario and new connection step

### PR DESCRIPTION
**Pull request by auarbekov**
**Issue #, if available:**
added t20 scenario and added new connection step

**Description of changes:**
- Add T20 scenario 
- Add new failed connection step

**Why is this change necessary:**
to implement Mqtt 5.0 features

**How was this change tested:**
Scenarios run manually and on CI

**Test results:**
```
Given my device is registered as a Thing....................................passed
And my device is running Greengrass.........................................passed
When I create a Greengrass deployment with components.......................passed
And I create client device "clientDeviceTest"...............................passed
When I associate "clientDeviceTest" with ggc................................passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:.passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.client.Mqtt5JavaSdkClient configuration to:.passed
And I deploy the Greengrass deployment configuration........................passed
Then the Greengrass deployment is COMPLETED on the device after 5 minutes...passed
And the aws.greengrass.clientdevices.mqtt.EMQX log on the device contains the line "is running now!." within 1 minutes.passed
And I discover core device broker as "default_broker" from "clientDeviceTest" in OTF.passed
And I connect device "clientDeviceTest" on aws.greengrass.client.Mqtt5JavaSdkClient to "default_broker" using mqtt "v5".passed
When I create a Greengrass deployment with components.......................passed
And I deploy the Greengrass deployment configuration........................passed
Then the Greengrass deployment is COMPLETED on the device after 299 seconds.passed
And I discover core device broker as "default_broker" from "clientDeviceTest" in OTF.passed
And I force to set broker "default_broker" with port 9443...................passed
Then I wait 60 seconds......................................................passed
And I can not connect device "clientDeviceTest" on aws.greengrass.client.Mqtt5JavaSdkClient to "default_broker" using mqtt "v5".passed
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 1]
```diff
@@ -820,6 +843,25 @@ public void discoverCoreDeviceBroker(String brokerId) {
...
     }
 
+
+    /**
+     * Set up IoT core broker port.
+     *
+     * @param brokerId broker name in tests
+     * @param port broker port in tests
+     */
+    @And("I force to set broker {string} with port {int}")
```
Comment (MikeDombo) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: why is this needed?

Comment (MikeDombo) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: Why isn't this using discovery like our other tests? Our internal testing does not do this. It adds the proper connectivity information with localhost and the custom port.

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: It that scenario we check case when client can connect to custom MQTT port AND can't connect to wrong port. For that we use discovery as usually but after that rewrite port in discovered list of  address information  to wrong port. In Evergreen similar thing did by other way.

Comment (MikeDombo) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: I'm looking at GGAD-1-T20 right now. It does not do this, it is changing the discovery information. What step do you claim is doing this?

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: That inject invalid connectivity information for non-yet actual port number 9001 which resulting second client failed to connect on port 9001.

With OTF we have no step like ''add connectivity information with host "localhost" and port "9001" we use IPdetector instead and it can't send fake port to backend.

Comment (MikeDombo) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: And why don't you have step for that?

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: We can try to implement  by reconfigure IPDetector but I expecting rice between components and scenario. 
We already got a lot of such issues and usually resolved it by adding steps for waiting until string appears in log file. With IPDetector it be hardly possible because IPDetector report action to log several times. Because we can stuck on other waiting step it hard to predict how many times IPDetector send information to backend and  how many entries about it will be logged.

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: We just combine both methods.

Comment (MikeDombo) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: Please update this as suggested to set the connectivity info.

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:853]: Ok, Akezhan will continue on Monday due to national holidays.

**Review by MikeDombo — COMMENTED** [review]
Is this ready for review?

**Comment by bgklika** [comment]
Under internal review, I do not sure it is ready for your review. Probably close to finish.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 1 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: Fixed in latest commits

**Review by MikeDombo — APPROVED** [review]
